src/test/java/com/exasol/adapter/dialects/rls/RowLevelSecurityQueryRewriterTest.java

```diff
@@ -0,0 +1,148 @@
+        this.properties = new AdapterProperties(this.rawProperties);
+        this.connectionMock = mock(Connection.class);
+        when(this.connectionMock.getMetaData()).thenReturn(this.databaseMetaDataMock);
+        when(this.connectionMock.prepareStatement(ArgumentMatchers.any())).thenReturn(preparedStatementMock);
+        setConnectionNameProperty();
+        this.metadataReader = new ExasolMetadataReader(this.connectionMock, this.properties);
+    }
+
+    @Test
+    void testRewriteUnprotectedTables() throws SQLException, AdapterException {
+        final SqlColumn left = new SqlColumn(1,
+                ColumnMetadata.builder().name("amount").type(DataType.createDecimal(20, 0)).build());
+        final SqlLiteralExactnumeric right = new SqlLiteralExactnumeric(BigDecimal.valueOf(2));
+        final SqlStatementSelect statement = createSelectStatement().whereClause(new SqlPredicateEqual(left, right))
+                .build();
+        final RowLevelSecurityQueryRewriter rewriter = new RowLevelSecurityQueryRewriter(this.dialect,
+                this.metadataReader, this.connectionMock);
+        assertThat(rewriter.rewrite(statement, this.exaMetadata, this.properties),
+                containsString("STATEMENT 'SELECT \"item\" FROM \"order_items\" WHERE \"amount\" = 2'"));
+    }
+
+    @Test
+    void testRewriteRolesWithoutWhereClause() throws SQLException, AdapterException {
+        final ResultSet resultSetColumnsMock = mock(ResultSet.class);
+        when(resultSetColumnsMock.next()).thenReturn(true);
+        when(this.databaseMetaDataMock.getColumns(any(), any(), any(), eq("EXA_ROW_ROLES")))
+                .thenReturn(resultSetColumnsMock);
+        final SqlStatementSelect statement = createSelectStatement().build();
+        final RowLevelSecurityQueryRewriter rewriter = new RowLevelSecurityQueryRewriter(this.dialect,
+                this.metadataReader, this.connectionMock);
+        assertThat(rewriter.rewrite(statement, this.exaMetadata, this.properties), containsString(
+                "STATEMENT 'SELECT \"item\" FROM \"order_items\" WHERE BIT_AND(\"EXA_ROW_ROLES\", 9223372036854775811) <> 0'"));
+    }
+
+    @Test
+    void testRewriteRolesWithSimpleWhereClause() throws SQLException, AdapterException {
+        final ResultSet resultSetColumnsMock = mock(ResultSet.class);
+        when(resultSetColumnsMock.next()).thenReturn(true);
+        when(this.databaseMetaDataMock.getColumns(any(), any(), any(), eq("EXA_ROW_ROLES")))
+                .thenReturn(resultSetColumnsMock);
+        final SqlColumn left = new SqlColumn(1,
+                ColumnMetadata.builder().name("amount").type(DataType.createDecimal(20, 0)).build());
+        final SqlLiteralExactnumeric right = new SqlLiteralExactnumeric(BigDecimal.valueOf(2));
+        final SqlStatementSelect statement = createSelectStatement().whereClause(new SqlPredicateEqual(left, right))
```

Comment:
There is a simpler way to express this:

`whereClause(eq(column("amount"), 2))`

Reply:
This is not `sql-statement-builder`